homeassistant/components/melcloud/sensor.py

```diff
@@ -207,9 +203,20 @@ def __init__(
             description.key = f"{description.key}-zone-{zone.zone_index}"
         super().__init__(api, description)
         self._zone = zone
-        self._attr_name = f"{api.name} {zone.name} {description.name}"
 
     @property
-    def native_value(self):
+    def device_info(self) -> DeviceInfo:
+        """Return device specific attributes."""
+        api = self._api.device
+        return DeviceInfo(
+            identifiers={(DOMAIN, f"{api.mac}-{api.serial}-{self._zone.zone_index}")},
+            connections={(CONNECTION_NETWORK_MAC, api.mac)},
+            manufacturer="Mitsubishi Electric",
+            model="ATW zone device",
+            name=f"{self._zone.name} {self._api.name}",
+        )
```

Comment:
Dont forget to add `via_device` with this attribute you can link this zone to the device that actually provides these zones.

Reply:
Uhm, looking better I think that this dedicated device should be created also in climate entity. Give me some moments

Reply:
Ok, now looks good.